script_library/openstack-preflight.py

```diff
@@ -0,0 +1,158 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+"""
+Connect to an OpenStack cloud and then test to make sure
+that all of our required configs exist.
+
+This script loads vars/deploy-on-openstack.yml to check against.
+
+"""
+
+import argparse
+import os
+
+import openstack
+from openstack.config import loader
+import sys
+import yaml
+
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-f", "--file",
+                    help="The yaml file for vars input")
+parser.add_argument("-s", "--ses",
+                    help="Run the ses checks?",
+                    default=False, action="store_true")
+parser.add_argument("-c", "--caasp",
+                    help="Run the caasp checks?",
+                    default=False, action="store_true")
+parser.add_argument("-o", "--osh",
+                    help="Run the osh checks?",
+                    default=False, action="store_true")
+parser.add_argument("-a", "--all",
+                    help="Run all the checks?",
+                    default=False, action="store_true")
+
+CLOUD = os.getenv('OS_TEST_CLOUD', 'devstack')
```

Comment:
Could we use OS_CLOUD here instead of creating a new variable?

Reply:
done